### DISCUSSION TITLE
Safe deployments

### DISCUSSION THREAD

**Discussion by fcrios145** [pocketbase/pocketbase#5026] [Q&A] [answered]
I'm about to going to production with Pocketbase and I was thinking about this
What it's a safest way of doing deployments to the server

According to what Gani said

if I'm not using anything like nginx or docker and I will be serving my app as a standalone binary.
Then stopping the binary execution and then replacing the bin and then restating the service will be enough right ?

**Comment by ganigeorgiev** [comment, marked as answer]
> Then stopping the binary execution and then replacing the bin and then restating the service will be enough right ?

Yes, that should be enough.

On Linux it is also possible to replace the old binary while the old one is still running when using `mv`, `rename` or any other command that will unlink the old one as part of its operation.
`rsync` (without! the `--in-place` flag) also should be safe. For [Presentator](https://github.com/presentator/presentator) I usually have a short `deploy.sh` that I just invoke when I have to deploy a new version of the app. Parameterized this could look like:

```sh
#!/bin/bash

VPS_IP="..."
SERVICE_NAME="presentator"
EXECUTABLE_NAME="presentator"
EXECUTABLE_PATH="/path/to/presentator"

# build executable (if you use PocketBase as Go framework)
go mod tidy
GOOS=linux GOARCH=arm64 CGO_ENABLED=0 go build -o $EXECUTABLE_NAME

# upload to the remote server
rsync -avz -e ssh ./$EXECUTABLE_NAME root@$VPS_IP:$EXECUTABLE_PATH

# ssh into the server and restart the service
ssh -T root@$VPS_IP << EOL
systemctl restart $SERVICE_NAME
EOL
```
(_`chmod +x ./deploy.sh` to make it runnable and then you should be able to just call it: `./deploy.sh`_)
